app/models/import/technologies/fetcher.rb

```diff
@@ -0,0 +1,53 @@
+class Import
+  module Technologies
+    class Fetcher
+      def initialize(etm_scenario)
+        @etm_scenario = etm_scenario
+        @scenario_id  = etm_scenario.fetch(:id)
+      end
+
+      def fetch
+        builders.flat_map do |builder|
+          builder.new(gqueries, @etm_scenario).build(technologies_from)
+        end
+      end
+
+      private
+
+      # Internal: Imports the requested data from the remote provider and returns
+      # the JSON response as a Hash.
+      def response
+        @response ||= EtEngineConnector.new(keys: keys)
+          .stats(@scenario_id)['nodes']
+      end
+
+      def keys
+        Technology.importable.map(&:key)
+      end
+
+      # Internal: Given a response, splits out the nodes into discrete technologies.
+      #
+      # Returns an array.
+      def technologies_from
+        @technologies_from ||= response.flat_map do |key, data|
+          TechnologyBuilder.build(key, data)
+        end
+      end
+
+      def gqueries
```

Comment:
I find that importing a new scenario still sends a lot of requests to ETEngine, albeit for the same Gqueries each time. Memoizing the Gqueries (`@gqueries ||= ...`) prevents the unnecessary extra requests.

#### Before

```
Started GET "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:30:59 +0100
Processing by Api::V3::ScenariosController#show as JSON
  Parameters: {"id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:30:59 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started POST "/api/v3/scenarios/493839/converters/stats" for 127.0.0.1 at 2016-05-18 12:31:00 +0100
Processing by Api::V3::ConvertersController#stats as JSON
  Parameters: {"keys"=>["energy_flexibility_p2g_electricity", "households_flexibility_p2h_electricity", "households_flexibility_p2p_electricity", "households_solar_pv_solar_radiation", "households_space_heater_combined_network_gas", "households_space_heater_heatpump_air_water_electricity", "households_space_heater_heatpump_ground_water_electricity", "households_space_heater_hybrid_heatpump_air_water_electricity", "households_space_heater_network_gas", "households_water_heater_heatpump_air_water_electricity", "households_water_heater_heatpump_ground_water_electricity", "households_water_heater_hybrid_heatpump_air_water_electricity", "transport_car_using_electricity"], "scenario_id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:31:01 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:31:02 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:31:03 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:31:04 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:31:05 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started POST "/api/v3/scenarios/493839/converters/stats" for 127.0.0.1 at 2016-05-18 12:31:09 +0100
Processing by Api::V3::ConvertersController#stats as JSON
  Parameters: {"keys"=>["households_flexibility_p2p_electricity"], "scenario_id"=>"493839"}
```

#### After

```
Started GET "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:41:38 +0100
Processing by Api::V3::ScenariosController#show as JSON
  Parameters: {"id"=>"493839"}

Started PUT "/api/v3/scenarios/493839" for 127.0.0.1 at 2016-05-18 12:41:38 +0100
Processing by Api::V3::ScenariosController#update as JSON
  Parameters: {"gqueries"=>["electric_cars_additional_costs", "etmoses_electricity_base_load_demand", "etmoses_hot_water_buffer_demand", "etmoses_space_heating_buffer_demand", "number_of_buildings", "present_demand_in_source_of_electricity_in_buildings"], "id"=>"493839"}

Started POST "/api/v3/scenarios/493839/converters/stats" for 127.0.0.1 at 2016-05-18 12:41:39 +0100
Processing by Api::V3::ConvertersController#stats as JSON
  Parameters: {"keys"=>["energy_flexibility_p2g_electricity", "households_flexibility_p2h_electricity", "households_flexibility_p2p_electricity", "households_solar_pv_solar_radiation", "households_space_heater_combined_network_gas", "households_space_heater_heatpump_air_water_electricity", "households_space_heater_heatpump_ground_water_electricity", "households_space_heater_hybrid_heatpump_air_water_electricity", "households_space_heater_network_gas", "households_water_heater_heatpump_air_water_electricity", "households_water_heater_heatpump_ground_water_electricity", "households_water_heater_hybrid_heatpump_air_water_electricity", "transport_car_using_electricity"], "scenario_id"=>"493839"}

Started POST "/api/v3/scenarios/493839/converters/stats" for 127.0.0.1 at 2016-05-18 12:41:44 +0100
Processing by Api::V3::ConvertersController#stats as JSON
  Parameters: {"keys"=>["households_flexibility_p2p_electricity"], "scenario_id"=>"493839"}
```

I'm not sure where that extra request for "households_flexibility_p2p_electricity" comes from. Maybe the front-end form since "Battery" is the default selection in the "Add technology" field?